streams/src/main/java/org/apache/kafka/streams/state/internals/CompositeReadOnlyWindowStore.java

```diff
@@ -43,45 +43,58 @@ public CompositeReadOnlyWindowStore(final StateStoreProvider provider,
         this.storeName = storeName;
     }
 
-    @Override
-    public WindowStoreIterator<V> fetch(final K key, final long timeFrom, final long timeTo) {
+    private interface Fetcher<K, V, IteratorType extends KeyValueIterator<?, V>> {
+        IteratorType fetch(ReadOnlyWindowStore<K, V> store);
+        IteratorType empty();
+    }
+
+    public <ITERATORTYPE extends KeyValueIterator<?, V>> ITERATORTYPE fetch(Fetcher<K, V, ITERATORTYPE> fetcher) {
```

Comment:
uggh - i think i saw you mentioned something about the checkstyle violation todo with generic types. Maybe we should change that rule? All caps is pretty ugly and Hard to read.